### PR TITLE
concord-server,console: remove master branch reference

### PR DESCRIPTION
**Pull request by amithkb**
Github nomenclature change needs to dereference default branch 'master'. 
Update to make repository `branch` as mandatory field and resolve while insertion and update if null

### CODE REVIEW

**Comment by ibodrov** [comment]
@concord-ci test it

File: server/db/src/main/resources/com/walmartlabs/concord/server/db/liquibase.xml [hunk 1]
```diff
@@ -91,5 +91,5 @@
     <include file="v1.60.0.xml" relativeToChangelogFile="true"/>
     <include file="v1.66.0.xml" relativeToChangelogFile="true"/>
     <include file="v1.69.0.xml" relativeToChangelogFile="true"/>
-
+    <include file="v1.70.0.xml" relativeToChangelogFile="true"/>
```
Comment (ibodrov) [inline server/db/src/main/resources/com/walmartlabs/concord/server/db/liquibase.xml:94]: It's going to be 1.75.0+ most likely

**Comment by amithkb** [comment]
@concord-ci test it

**Comment by amithkb** [comment]
@concord-ci test it

**Comment by ibodrov** [comment]
@concord-ci test it

File: it/server/src/test/java/com/walmartlabs/concord/it/server/ITConstants.java [hunk 2]
```diff
@@ -9,9 +9,9 @@
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
- *
```
Comment (ibodrov) [inline it/server/src/test/java/com/walmartlabs/concord/it/server/ITConstants.java:12]: A bit noisy, could you please remove those unnecessary changes?

File: cli/src/main/java/com/walmartlabs/concord/cli/Run.java [hunk 3]
```diff
@@ -100,6 +100,9 @@
     @Option(names = {"-v", "--verbose"}, description = "verbose output")
     boolean verbose = false;
 
+    @Option(names = {"--imports-version/branch"}, description = "default imports version/branch")
```
Comment (ibodrov) [inline cli/src/main/java/com/walmartlabs/concord/cli/Run.java:103]: I've never seen this format of CLI toggles/parameters with `/` in it. Maybe we can shorten it to `--default-import-version`? Make it more "GNU-ish"? I don't think we'd have other uses for this value, but it sounds more generic and can potentially be useful for other things.

File: server/dist/src/main/resources/concord-server.conf [hunk 4]
```diff
@@ -255,6 +255,9 @@ concord-server {
         disabledProcessors = [
             "dir"
         ]
+
+        # default branch for import
+        defaultBranch = "master"
```
Comment (ibodrov) [inline server/dist/src/main/resources/concord-server.conf:260]: Perhaps we should put `main` as the default value then?

**Review by ibodrov — CHANGES_REQUESTED** [review]

File: server/db/src/main/resources/com/walmartlabs/concord/server/db/v1.76.0.xml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
+
+    <changeSet id="1760000" author="amith.k.b@walmartlabs.com">
+        <sql>
+            update REPOSITORIES set REPO_BRANCH = 'master' where REPO_BRANCH is NULL
```
Comment (brig) [inline server/db/src/main/resources/com/walmartlabs/concord/server/db/v1.76.0.xml:9]: `and  commit_id is null`

File: it/server/src/test/java/com/walmartlabs/concord/it/server/ITConstants.java (hunk 2 above)
Comment (amithkb) [reply it/server/src/test/java/com/walmartlabs/concord/it/server/ITConstants.java]: yes removed :)

File: cli/src/main/java/com/walmartlabs/concord/cli/Run.java (hunk 3 above)
Comment (amithkb) [reply cli/src/main/java/com/walmartlabs/concord/cli/Run.java]: sure

File: server/db/src/main/resources/com/walmartlabs/concord/server/db/v1.76.0.xml (hunk 5 above)
Comment (amithkb) [reply server/db/src/main/resources/com/walmartlabs/concord/server/db/v1.76.0.xml]: sure

File: server/dist/src/main/resources/concord-server.conf (hunk 4 above)
Comment (amithkb) [reply server/dist/src/main/resources/concord-server.conf:260]: I think, currently within the organization, the plan to migrate/adoption of `main` is still in progress.  Hence the plan is to keep it configurable and later we will change it to main

**Comment by amithkb** [comment]
@concord-ci test it

File: console2/public/cfg.js [hunk 6]
```diff
@@ -9,5 +9,7 @@ window.concord = {
                 icon: 'github'
             }
         ]
-    }
+    },
+    defaultRepoBranch: 'master'
```
Comment (brig) [inline console2/public/cfg.js:13]: why we need this default value if branch is mandatory?

Comment (amithkb) [reply console2/public/cfg.js:13]: [here](https://github.com/walmartlabs/concord/blob/baf49a25360152125cc1fc1cc01e0af882982c5b/console2/src/components/molecules/GitHubLink/index.tsx#L79)  - when `commitId` is not present but `path` exists

Comment (brig) [reply console2/public/cfg.js:13]: hmm... commitId or branch is mandatory now. maybe add `branch` as param for GitHubLink?

**Comment by ibodrov** [comment]
@concord-ci test it

File: console2/public/cfg.js (hunk 6 above)
Comment (amithkb) [reply console2/public/cfg.js]: ok removed default value from console config

File: server/dist/src/main/resources/concord-server.conf [hunk 7]
```diff
@@ -418,6 +421,9 @@ concord-server {
 
         sshTimeoutRetryCount = 1
         sshTimeout = "10 minutes"
+
+        # default Git branch
+        defaultBranch = "master"
```
Comment (brig) [inline server/dist/src/main/resources/concord-server.conf:426]: why we need this default value if branch is mandatory? :)

Comment (amithkb) [reply server/dist/src/main/resources/concord-server.conf:426]: As either commitId or Branch is mandatory. Hence there is still possibility like branch may be null when `repositories` are created by giving only commitId. So i think, we would need this default value @ [1](https://github.com/walmartlabs/concord/blob/6d80f9d09bd10ffd578af5096211800c44111eda/server/impl/src/main/java/com/walmartlabs/concord/server/org/triggers/GithubTriggerEnricher.java#L88), [2](https://github.com/walmartlabs/concord/blob/6d80f9d09bd10ffd578af5096211800c44111eda/server/impl/src/main/java/com/walmartlabs/concord/server/process/pipelines/processors/RepositoryProcessor.java#L111) ... wdyt?

File: server/dist/src/main/resources/concord-server.conf (hunk 4 above)
Comment (benbroadaway) [reply server/dist/src/main/resources/concord-server.conf:260]: It can be `main` in the repo here, and then customized/overridden in a deployed `server.conf` as necessary.

**Comment by ibodrov** [comment]
@concord-ci test it

**Comment by ibodrov** [comment]
@concord-ci test it

File: cli/src/main/java/com/walmartlabs/concord/cli/Run.java [hunk 8]
```diff
@@ -100,6 +100,9 @@
     @Option(names = {"-v", "--verbose"}, description = "verbose output")
     boolean verbose = false;
 
+    @Option(names = {"--default-import-version"}, description = "default import version or repo branch")
+    String defaultVersion = "master";
```
Comment (ibodrov) [inline cli/src/main/java/com/walmartlabs/concord/cli/Run.java:104]: If we change the server.conf default value to `main`, we should probably change it here too.
I don't think it's going to cause any real issues for the current users - there aren't that many CLI uses at the moment :-)

**Review by brig — APPROVED** [review]

File: server/dist/src/main/resources/concord-server.conf (hunk 4 above)
Comment (amithkb) [reply server/dist/src/main/resources/concord-server.conf]: changed to main

File: cli/src/main/java/com/walmartlabs/concord/cli/Run.java (hunk 8 above)
Comment (amithkb) [reply cli/src/main/java/com/walmartlabs/concord/cli/Run.java]: changed to main

**Review by ibodrov — APPROVED** [review]
